src/main/java/jarvis/model/Task.java

```diff
@@ -74,10 +79,15 @@ public boolean equals(Object other) {
             return false;
         }
 
+
         Task otherTask = (Task) other;
-        return otherTask.getDesc().equals(getDesc())
-                && otherTask.getDeadline().equals(getDeadline())
-                && otherTask.isDone() == isDone();
+
+        boolean taskDeadlineEquality = taskDeadline == null || otherTask.taskDeadline == null
+                ? taskDeadline == null && otherTask.taskDeadline == null
+                : otherTask.taskDeadline.equals(taskDeadline);
```

Comment:
Same issue as the one mentioned in AddNotesCommand, might be better to separate the cases with if-else statements